### PR TITLE
python3Packages.jsonschema: 3.2.0 -> 4.3.3

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
https://github.com/Julian/jsonschema/releases

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/jsonschema/default.nix [hunk 1]
```diff
@@ -1,38 +1,59 @@
...
-  version = "3.2.0";
+  version = "4.3.3";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.7";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "c8a85b28d377cc7737e46e2d9f2b4f44ee3c0e1deac6bf46ddefc7187d30797a";
+    hash = "sha256-yKhbKNN3zHc35G4tnytPRO48Dh3qxr9G3e/HGH0weXo=";
```
Comment (xfix) [inline pkgs/development/python-modules/jsonschema/default.nix:25]: This checksum doesn't work for me.

```suggestion
    sha256 = "1phazxdpxf4agfl9nfalmdbgd1mmg7pch5axcfpyilay177d847j";
```

**Review by sugar700 — CHANGES_REQUESTED** [review]
Otherwise this looks fine. I wanted to package a program that needs a newer version of jsonschema, and with this change, it works fine. Note that I didn't check whether already existing programs break with this change or not.

**Comment by fabaff** [comment]
This has probably to go to `staging`. As far as I remember would it touch between 200 and 300 packages.

File: pkgs/development/python-modules/jsonschema/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/jsonschema/default.nix]: Should be fixed now. Thanks.

Comment (fabaff) [reply pkgs/development/python-modules/jsonschema/default.nix]: Or not, got again hash mismatch.

**Pull request closed without merging**